test/unit/modunit_pico_mld.c

```diff
@@ -118,16 +119,12 @@ END_TEST
 START_TEST(tc_pico_mld_report_expired)
 {
     struct mld_timer *t = PICO_ZALLOC(sizeof(struct mld_timer));
-    struct pico_ip6 zero = {{0}};
-    struct mcast_parameters p;
+    struct pico_ip6 zero = {0};
```

Comment:
I tried the above with a few different versions and got mixed results.
I updated to GCC 6.2.1 and this version didn't complain with `something = {{0}}` , `something = {0}` and `something = {}` (it accepts all three it seems). I tried with GCC 5.4.0 as well and this complains when I use `something = {{0}}`.
GCC version 6.1.1 complained when I used  `something = {{0}}`, but GCC version 6.2.1 doesn't anymore... But it is indeed weird that this sort of code in the stack doesn't give warnings.
If I compile with GCC 4.8.5, I also see warnings popping up again, I tried to fix them in 3f5f6ea363e1afabd844100b61ec90334535627e. @frederikvs Could you try compiling now and see if you get warnings? Maxime's suggestion seems to be valid across GCC versions.